annotation/tests/test_annotations.py

```diff
@@ -482,3 +588,491 @@ def test_construct_annotated_doc_db_error(
+):
+    expected_user_1 = page_revision_list_all[1][0]["user_id"]
+    expected_page_rev_1 = {**page_revision_list_all[1][0]}
+    expected_user_2 = page_revision_list_all[2][0]["user_id"]
+    expected_page_rev_2 = {**page_revision_list_all[2][0]}
+    expected_pages = {1: [], 2: []}
+
+    with patch(
+        "annotation.annotations.main.convert_bucket_name_if_s3prefix",
+        return_value=TEST_TENANT,
+    ) as mock_convert_prefix, patch(
+        "annotation.annotations.main.load_page"
+    ) as mock_load_page:
+        load_all_revisions_pages(page_revision_list_all, TEST_TENANT)
+        mock_convert_prefix.assert_called_once_with(TEST_TENANT)
+        mock_load_page.assert_has_calls(
+            [
+                call(
+                    None,
+                    [],
+                    TEST_TENANT,
+                    1,
+                    expected_user_1,
+                    expected_page_rev_1,
+                    True,
+                ),
+                call(
+                    None,
+                    [],
+                    TEST_TENANT,
+                    2,
+                    expected_user_2,
+                    expected_page_rev_2,
+                    True,
+                ),
+            ]
+        )
+        assert mock_load_page.call_count == 2
+    assert page_revision_list_all == expected_pages
+
+
+def test_load_latest_revs_pages(
+    page_revision_list_latest: Dict[int, Dict[str, LatestPageRevision]]
+):
+    expected_user_1 = list(page_revision_list_latest[1].keys())[0]
+    expected_user_2 = list(page_revision_list_latest[2].keys())[0]
+    expected_page_rev_1 = {**page_revision_list_latest[1][expected_user_1]}
+    expected_page_rev_2 = {**page_revision_list_latest[2][expected_user_2]}
+    expected_pages = {1: [], 2: []}
+
+    with patch(
+        "annotation.annotations.main.convert_bucket_name_if_s3prefix",
+        return_value=TEST_TENANT,
+    ) as mock_convert_prefix, patch(
+        "annotation.annotations.main.load_page"
+    ) as mock_load_page:
+        load_latest_revision_pages(page_revision_list_latest, TEST_TENANT)
+        mock_convert_prefix.assert_called_once_with(TEST_TENANT)
+        mock_load_page.assert_has_calls(
+            [
+                call(
+                    None,
+                    [],
+                    TEST_TENANT,
+                    1,
+                    expected_user_1,
+                    expected_page_rev_1,
+                    True,
+                ),
+                call(
+                    None,
+                    [],
+                    TEST_TENANT,
+                    2,
+                    expected_user_2,
+                    expected_page_rev_2,
+                    True,
+                ),
+            ]
+        )
+        assert mock_load_page.call_count == 2
+    assert page_revision_list_latest == expected_pages
+
+
+def test_load_annotated_pages_for_particular_rev(
```

Comment:
Function utilize a logger, you can mock the logger to verify that the correct messages are being logged.